src/redux/slices/persistGateHydration.ts

```diff
@@ -0,0 +1,56 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { REHYDRATE } from 'redux-persist';
+
+import { RootState } from '../RootState';
+
+import SliceName from '@/redux/types/SliceName';
+
+export type PersistGateHydrationState = {
+  isPersistGateHydrationInProgress: boolean;
+  isPersistGateHydrationComplete: boolean;
+};
+
+const initialState: PersistGateHydrationState = {
+  isPersistGateHydrationInProgress: false,
+  isPersistGateHydrationComplete: false,
```

Comment:
Quick question, do we need 2 states? cannot we just set only 1 state since from what I am getting, we only use either one of them and not both at the same time.

Reply:
Good question, my initial implementation involved a single state. 

tl;dr because the redux store is synchronous, you can't tell it do dispatch an action *AFTER* one occurs. You'll need to dispatch it separately after. 

The `isPersistGateHydrationInProgress` flags that the Hydration is in progress/done, which triggers a useEffect that sets `isPersistGateHydrationComplete` to true. 

The application logic will be using the `isPersistGateHydrationComplete`. Let me know if it makes sense.